lib/discoverable.coffee

```diff
@@ -115,7 +112,7 @@ determineServiceInfo = (service) ->
 	info = {}
 
 	types = service.service.match(/^(_(.*)\._sub\.)?_(.*)\._(.*)$/)
-	if types[1] is undefined and types[2] is undefined
+	if not types[1]? and not types[2]?
```

Comment:
Will still throw with `cannot read property '1' of null` TypeError if `.match()` returns `null`. This could instead be:

```coffeescript
if not types?[1] and not types?[2]
```